pandas/core/arrays/period.py

```diff
@@ -185,6 +333,55 @@ def _generate_range(cls, start, end, periods, freq, fields):
 
         return subarr, freq
 
+    @classmethod
+    def _concat_same_type(cls, to_concat):
+        freq = {x.freq for x in to_concat}
+        assert len(freq) == 1
+        freq = list(freq)[0]
+        values = np.concatenate([x._data for x in to_concat])
+        return cls._from_ordinals(values, freq=freq)
+
+    @property
+    def asi8(self):
+        return self._ndarray_values.view('i8')
```

Comment:
Isn’t this already in the datetimelikemixin?

Reply:
I think we should also look if this property is actually needed. Where is it used? In things that are common to datetime like arrays?

Because internally here, in many places we also use `_data`, which is basically the same?

For DatetimeArray, there is of course a clearer difference between `_data` and `asi8` (where that would be datetime64 vs int64, while here both are int64). That might be enough reason to keep to share code between the datetimelikes.

And we also have `_ndarray_values` that is used in some places here in the class. 

I think for PeriodArray-only internal implementations, we should maybe mainly be using `_data` ?